VENDOR_NEUTRALITY.md

```diff
@@ -0,0 +1,33 @@
+# Metal3-io vendor neutrality declaration
+
+As a Cloud Native Computing Foundation project, Metal3-io will keep its
```

Comment:
Is "Metal3-io" our canonical name in CNCF? We definitely use just "Metal3" everywhere in our documentation.

Reply:
It is actually something I was confused a bit so I had too look it up after you commented and:

https://github.com/cncf/foundation/blob/main/project-maintainers.csv#L688  here we are metal3-io
here we are metal3 https://landscape.cncf.io/?item=provisioning--automation-configuration--metal
here the name of the project os Metal3 but the name of the PR is metal3-io:
https://github.com/cncf/toc/pull/408/files
here it is Metal3.io
https://github.com/cncf/toc/issues/1365

I think the official name is Metal3 because that was used to join CNCF and I think once we agreed to always refer to the project as Metal3 so I will fix it in the doc, Thanks for pointing it out!